### ISSUE TITLE
surface issue since latest GR

### ISSUE THREAD

**Issue by t-bltg** [jheinen/GR.jl#507]
Our test example at https://docs.juliaplots.org/latest/gallery/gr/generated/gr-ref050/#gr_ref050 is failing CI ([see here](https://github.com/JuliaPlots/Plots.jl/actions/runs/4147879616/jobs/7175262366)) since GR `0.71.6`.

I first thought this was some minor changes causing CI failure (small comparison tolerance), but this is the example output, so I think there is a clear regression:

![ref050](https://user-images.githubusercontent.com/13423344/218252363-9103ba66-c562-4882-a0d6-60826da27841.png)

Note that this wasn't caught by https://github.com/jheinen/GR.jl/blob/master/test/downstream_test.jl, since we don't test image correctness there, only that they run without bugs / crashes.

cc @jheinen

**Comment by jheinen** [comment]
Unfortunately, I cannot reproduce the problem with `Plots 1.38.5` and `GR 0.71.6`.

<img width="714" alt="Screenshot 2023-02-11 at 17 29 01" src="https://user-images.githubusercontent.com/5718244/218269401-9b8f7977-ed32-4372-a860-d95db278ad55.png">

Or does the problem only occur when executed as part of the `Plots` examples to with a specific output device?

**Comment by t-bltg** [comment]
Weird, I can reproduce this on `linux`, `julia 1.9.0-beta4` (shouldn't matter, fails all `julia` versions in CI):

```julia
pkg> st -m Plots GR
Status `[...]/.julia/environments/v1.9/Manifest.toml`
  [28b8d3ca] GR v0.71.6
  [91a5bcdd] Plots v1.38.5
julia> using Plots
julia> Plots.test_examples(50)
[ Info: Testing plot: gr:50:3D surface with axis guides
```
Screenshot of the GKS QtTerm output:
![surf](https://user-images.githubusercontent.com/13423344/218270603-5ed666ec-b979-4a61-b822-5a1079cf2eb0.png)

**Comment by t-bltg** [comment]
Side note, maybe unrelated, but I now get systematic errors when exiting julia after plotting:
```julia
julia> exit(0)
X Error:  BadDrawable
  Request Major code 55 ()
  ResourceID 0x0
  Error Serial #33
  Current Serial #39
```
(related to https://github.com/jheinen/GR.jl/issues/480)

**Comment by t-bltg** [comment]
I can confirm that downgrading to `GR@0.71.5` works, so clearly the new version is at fault:
```julia
$ julia
pkg> add GR@0.71.5
   Resolving package versions...
   Installed GR_jll ─ v0.71.5+0
   Installed GR ───── v0.71.5
  Downloaded artifact: GR
    Updating `[...]/.julia/environments/v1.9/Project.toml`
⌃ [28b8d3ca] ↓ GR v0.71.6 ⇒ v0.71.5
    Updating `[...]/.julia/environments/v1.9/Manifest.toml`
⌃ [28b8d3ca] ↓ GR v0.71.6 ⇒ v0.71.5
⌅ [d2c73de3] ↓ GR_jll v0.71.6+0 ⇒ v0.71.5+0
Precompiling environment...
  4 dependencies successfully precompiled in 148 seconds. 558 already precompiled.
  1 dependency had warnings during precompilation:
┌ Plots [91a5bcdd-55d7-5caf-9e0b-520d859cae80]
│  [ Info: GR
└  
julia> using Plots
julia> Plots.test_example(50)  # correct previous surface
julia> exit(0)  # no error (https://github.com/jheinen/GR.jl/issues/507#issuecomment-1426831771)
$ echo 'all good'
```

**Comment by jheinen** [comment]
Unfortunately, [this](https://github.com/sciapp/gr/commit/b33e887dac4f248b37e708c508b2da59c9474aec) commit has a bug built into it that was not noticed in our CI tests because we had software rendering turned on in the CI run (`GR3_USE_SR=1`) and also use these setting on all our systems.

We will correct this ASAP and include more CI tests so that something like this does not happen again.
Sorry for the confusion.

**Comment by t-bltg** [comment]
Thanks !

**Comment by jheinen** [comment]
The problem is fixed in 0.71.7.
